src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -178,4 +211,526 @@ export class NetworkAreaDiagramViewer {
         const viewbox: VIEWBOX = svg.viewBox.baseVal;
         return { width: width, height: height, viewbox: viewbox };
     }
+
+    private enablePanzoom() {
+        this.svgDraw?.panZoom({
+            panning: true,
+            zoomMin: 0.5 / this.ratio,
+            zoomMax: 30 * this.ratio,
+            zoomFactor: 0.15,
+            margins: { top: 0, left: 0, right: 0, bottom: 0 },
+        });
+    }
+
+    private disablePanzoom() {
+        this.svgDraw?.panZoom({
+            panning: false,
+        });
+    }
+
+    private startDrag(event: Event) {
+        const draggableElem = DiagramUtils.getDraggableFrom(
+            event.target as SVGElement
+        );
+        if (!draggableElem) {
+            return;
+        }
+        this.disablePanzoom(); // to avoid panning the whole SVG when moving a node
+        this.ctm = this.svgDraw?.node.getScreenCTM(); // used to compute mouse movement
+        this.selectedElement = draggableElem as SVGGraphicsElement; // element to be moved
+        this.selectedElement.style.cursor = 'grabbing';
+        this.initialPosition = DiagramUtils.getPosition(this.selectedElement); // used for the offset
+    }
+
+    private drag(event: Event) {
+        if (this.selectedElement) {
+            event.preventDefault();
+            this.updateGraph(event);
+            this.initialPosition = DiagramUtils.getPosition(
+                this.selectedElement
+            );
+        }
+    }
+
+    private endDrag(event: Event) {
+        if (this.selectedElement) {
+            this.updateGraph(event);
+            //this.selectedElement.style.cursor = 'grab';
+            this.selectedElement.removeAttribute('style');
+            this.selectedElement = null;
+            this.initialPosition = new Point(0, 0);
+            this.enablePanzoom();
+        }
+    }
+
+    // position w.r.t the SVG box
+    private getMousePosition(event: MouseEvent): Point {
+        return new Point(
+            (event.clientX - (this.ctm?.e ?? 0)) / (this.ctm?.a ?? 1),
+            (event.clientY - (this.ctm?.f ?? 0)) / (this.ctm?.d ?? 1)
+        );
+    }
+
+    private updateGraph(event: Event) {
+        const offset = this.getMousePosition(event as MouseEvent);
+        this.moveNode(offset);
+        this.moveNodeText(offset);
+        this.moveEdges();
+    }
+
+    private moveNode(offset: Point) {
+        //this.transform = DiagramUtils.getTransform(this.selectedElement);
+        //this.transform?.setTranslate(offset.x, offset.y);
+        this.selectedElement?.setAttribute(
+            'transform',
+            'translate(' + offset.x.toFixed(2) + ',' + offset.y.toFixed(2) + ')'
+        );
+    }
+
+    private moveNodeText(offset: Point) {
+        const translation = new Point(
+            offset.x - this.initialPosition.x,
+            offset.y - this.initialPosition.y
+        );
+        // move node text
+        this.moveTextNode(this.selectedElement?.id + '-textnode', translation);
```

Comment:
moveTextNode vs moveNodeText is not very clear, rename it to make it clearer:
- moveNodeText -> simply moveText?
- moveTextNode -> moveSvgElement? indeed, there's nothing about text in there (except the variable names)

Reply:
methods renamed